gbs-occt/CMakeLists.txt

```diff
@@ -1,27 +1,21 @@
 cmake_minimum_required(VERSION 3.17.0)
 
-project(occt LANGUAGES CXX)
-# 
+project(gbs-occt LANGUAGES CXX)
+
 # gbslib
-#
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../)
-# 
-# set files fo lib
-# 
+include_directories(${CMAKE_SOURCE_DIR})
+
 file(GLOB OCCT_UTILS_SRC_LIST "*.cpp")
-add_library(gbs-${PROJECT_NAME} SHARED ${OCCT_UTILS_SRC_LIST})
-target_link_libraries(gbs-${PROJECT_NAME} ${OpenCASCADE_LIBRARIES})
+add_library(${PROJECT_NAME} SHARED ${OCCT_UTILS_SRC_LIST})
+target_link_libraries(${PROJECT_NAME} ${OpenCASCADE_LIBRARIES})
+
+install(TARGETS ${PROJECT_NAME})
```

Comment:
In order to generate configuration files can you use this way to configure install?
```cmake
cmake_minimum_required(VERSION 3.17.0)

project(gbs-occt LANGUAGES CXX)

# gbslib
include_directories(${CMAKE_SOURCE_DIR})

set(TARGET_NAME OCCT)
file(GLOB OCCT_UTILS_SRC_LIST "*.cpp")
add_library(${TARGET_NAME} SHARED ${OCCT_UTILS_SRC_LIST})
target_link_libraries(${TARGET_NAME} ${OpenCASCADE_LIBRARIES})
set_target_properties(${TARGET_NAME} PROPERTIES  OUTPUT_NAME ${PROJECT_NAME})
install(TARGETS ${TARGET_NAME}
        EXPORT  ${PROJECT_NAME}Targets
)
install(
    EXPORT  ${PROJECT_NAME}Targets
    FILE ${PROJECT_NAME}.cmake
    NAMESPACE GBS::
    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
)
INSTALL_HEADERS(${PROJECT_NAME})

# tests
# add current dir for include in tests
include_directories(${CMAKE_CURRENT_SOURCE_DIR})
if(${GBS_BUILD_TESTS})
    add_subdirectory(tests)
endif(${GBS_BUILD_TESTS})
```

Reply:
yeap sure, it was the intent.
It also requires extra information. I'm not a pro of CMake but will make a try!

Reply:
Well I tried with  this code and it works like a charm ;)

Reply:
Do you want to be able to use `find_package` command?
Do you have an example of another project where you are using gbs modules and commands required to use them?
Thanks!

Reply:
Sorry I saw your response just now.

Does `project(gbs-occt LANGUAGES CXX)` still need to be rolledback to have `occt` as project name and I image `GBS::occt` module?

Reply:
Not with this project, but with others yes.
I made some modification on your branch and the files are correctly generated.
```cmake
# Generated by CMake

if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 2.5)
   message(FATAL_ERROR "CMake >= 2.6.0 required")
endif()
cmake_policy(PUSH)
cmake_policy(VERSION 2.6...3.17)
#----------------------------------------------------------------
# Generated CMake target import file.
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Protect against multiple inclusion, which would fail when already imported targets are added once more.
set(_targetsDefined)
set(_targetsNotDefined)
set(_expectedTargets)
foreach(_expectedTarget GBS::OCCT)
  list(APPEND _expectedTargets ${_expectedTarget})
  if(NOT TARGET ${_expectedTarget})
    list(APPEND _targetsNotDefined ${_expectedTarget})
  endif()
  if(TARGET ${_expectedTarget})
    list(APPEND _targetsDefined ${_expectedTarget})
  endif()
endforeach()
if("${_targetsDefined}" STREQUAL "${_expectedTargets}")
  unset(_targetsDefined)
  unset(_targetsNotDefined)
  unset(_expectedTargets)
  set(CMAKE_IMPORT_FILE_VERSION)
  cmake_policy(POP)
  return()
endif()
if(NOT "${_targetsDefined}" STREQUAL "")
  message(FATAL_ERROR "Some (but not all) targets in this export set were already defined.\nTargets Defined: ${_targetsDefined}\nTargets not yet defined: ${_targetsNotDefined}\n")
endif()
unset(_targetsDefined)
unset(_targetsNotDefined)
unset(_expectedTargets)


# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()

# Create imported target GBS::OCCT
add_library(GBS::OCCT SHARED IMPORTED)

set_target_properties(GBS::OCCT PROPERTIES
  INTERFACE_LINK_LIBRARIES "TKernel;TKMath;TKG2d;TKG3d;TKGeomBase;TKBRep;TKGeomAlgo;TKTopAlgo;TKPrim;TKBO;TKShHealing;TKBool;TKHLR;TKFillet;TKOffset;TKFeat;TKMesh;TKXMesh;TKService;TKV3d;TKOpenGl;TKMeshVS;TKIVtk;TKCDF;TKLCAF;TKCAF;TKBinL;TKXmlL;TKBin;TKXml;TKStdL;TKStd;TKTObj;TKBinTObj;TKXmlTObj;TKVCAF;TKXSBase;TKSTEPBase;TKSTEPAttr;TKSTEP209;TKSTEP;TKIGES;TKXCAF;TKXDEIGES;TKXDESTEP;TKSTL;TKVRML;TKXmlXCAF;TKBinXCAF;TKRWMesh"
)

if(CMAKE_VERSION VERSION_LESS 2.8.12)
  message(FATAL_ERROR "This file relies on consumers using CMake 2.8.12 or greater.")
endif()

# Load information for each installed configuration.
get_filename_component(_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
file(GLOB CONFIG_FILES "${_DIR}/gbs-occt-*.cmake")
foreach(f ${CONFIG_FILES})
  include(${f})
endforeach()

# Cleanup temporary variables.
set(_IMPORT_PREFIX)

# Loop over all imported files and verify that they actually exist
foreach(target ${_IMPORT_CHECK_TARGETS} )
  foreach(file ${_IMPORT_CHECK_FILES_FOR_${target}} )
    if(NOT EXISTS "${file}" )
      message(FATAL_ERROR "The imported target \"${target}\" references the file
   \"${file}\"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   \"${CMAKE_CURRENT_LIST_FILE}\"
but not all the files it references.
")
    endif()
  endforeach()
  unset(_IMPORT_CHECK_FILES_FOR_${target})
endforeach()
unset(_IMPORT_CHECK_TARGETS)

# This file does not depend on other imported targets which have
# been exported from the same project but in a separate export set.

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
cmake_policy(POP)

```

Reply:
To answer your previous question, yes, as well as gbs-render, and pass the include dir name as argument.
But if you want I can do it after merging your changes.

Reply:
Sure!
Thanks